authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,46 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
+
+from django.conf import settings
+
+from rest_framework import authentication, exceptions
+
+from .models import User
 
 """Configure JWT Here"""
+class JWTAuthentication(authentication.BaseAuthentication):
+    authentication_header_prefix = 'token'
+    def authenticate(self, request):
+        request.user = None
+        
+        auth_header = authentication.get_authorization_header(request).split()
+        auth_header_prefix = self.authentication_header_prefix.lower()
+
+        if not auth_header:
+            return None
+        if len(auth_header) == 1:
+            return None
+        elif len(auth_header) > 2:
+            return None
+        prefix = auth_header[0].decode('utf-8')
+
+        token = auth_header[1].decode('utf-8')
+        if prefix.lower() != auth_header_prefix:
+            return None
+        return self._authenticate_credentiatials(request, token)
+    
+    def _authenticate_credentiatials(self, request, token):
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY)
+        except:
+            msg = 'Invalid authentication. Could not decode token'
+            raise exceptions.AuthenticationFailed(msg)
+        try:
```

Comment:
@VivianDoreen Consider having a look at all the try block here and if they necessary and refactor them  or you leave a comment stating why you needed all the stated scenarios.

Reply:
ok, let me do that